js/src/components/SchemaSettings.js

```diff
@@ -34,17 +43,32 @@ class SchemaSettings extends Component {
 	 * @returns {boolean} True if a value differs from the initial value.
 	 */
 	shouldShowAlert() {
-		if ( this.props.pageType && this.props.pageType.value !== this.initialPageType ) {
+		if ( this.state.schema_page_type !== this.initialPageType ) {
 			return true;
 		}
 
-		if ( this.props.articleType && this.props.articleType.value !== this.initialArticleType ) {
+		if ( this.state.schema_article_type !== this.initialArticleType ) {
 			return true;
 		}
 
 		return false;
 	}
 
+	/**
+	 * Changes the state appropriately whenever an option is highlighted in a specific field.
+	 * This is needed because we need to show an alert when the select is "dirty", which is before the onBlur event sometimes.
+	 *
+	 * @param {string} fieldName      The name of the field to which this option belongs
+	 * @param {string} focussedOption The name of the focussed option.
+	 *
+	 * @returns {void}
+	 */
+	handleOptionFocus( fieldName, focussedOption ) {
+		this.setState( {
```

Comment:
You could move the `shouldShowAlert` to the state, and set the value here directly.
`shouldShowAlert` was there before to check props on the fly. But now that there is state might as well save the thing you actually use in the render.

Something like:
```js
let showAlert = false;

if ( fieldName === "schema_page_type" && focusedOption !== this.initialPageType ) {
	showAlert = true;
}

if ( fieldName === "schema_article_type" && focusedOption !== this.initialArticleType ) {
	showAlert = true;
}

this.setState( { showAlert } );
```

Reply:
I tried to do this, but the problem is that if you then reset _ONE_ of the fields to the initial value, if will fall through the if statements and set the state to false for _BOTH_ fields.

Of course you can catch this with more elaborate if statements, but I think the non-state manner we have currently works better.

Reply:
I'll explain in a call if you want 😓